libbeat/processors/decode_xml/decode_xml_test.go

```diff
@@ -0,0 +1,482 @@
+				"message": 1,
+			},
+			error:        true,
+			errorMessage: "field value is not a string",
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.description, func(t *testing.T) {
+			t.Parallel()
+			f := &decodeXML{
+				log:    logp.NewLogger("decode_xml"),
+				config: test.config,
+			}
+
+			event := &beat.Event{
+				Fields: test.Input,
+			}
+			newEvent, err := f.Run(event)
+			if !test.error {
+				assert.NoError(t, err)
+			} else {
+				assert.Error(t, err)
+				assert.EqualError(t, err, test.errorMessage)
+			}
+			assert.Equal(t, test.Output, newEvent.Fields)
+		})
+	}
+}
+
+// BenchmarkProcessor_Run/single_object-4             96710             11232 ns/op            8594 B/op        158 allocs/op
+// BenchmarkProcessor_Run/nested_and_array_object-4   42454             28094 ns/op           20296 B/op        376 allocs/op
+func BenchmarkProcessor_Run(b *testing.B) {
+	c := defaultConfig()
+	p, err := newDecodeXML(c)
+
+	b.Run("single_object", func(b *testing.B) {
+		evt := &beat.Event{Fields: map[string]interface{}{
+			"message": `<?xml version="1.0"?>
+				<catalog>
+					<book>
+					<author>William H. Gaddis</author>
+					<title>The Recognitions</title>
+					<review>One of the great seminal American novels of the 20th century.</review>
+				</book>
+				</catalog>`,
+		}}
+
+		for i := 0; i < b.N; i++ {
+			_, err = p.Run(evt)
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+
+	b.Run("nested_and_array_object", func(b *testing.B) {
+		evt := &beat.Event{Fields: map[string]interface{}{
+			"message": `<?xml version="1.0"?>
+				<catalog>
+					<book>
+					<author>William H. Gaddis</author>
+					<title>The Recognitions</title>
+					<review>One of the great seminal American novels of the 20th century.</review>
+				</book>
+				<book>
+					<author>Ralls, Kim</author>
+					<title>Midnight Rain</title>
+					<review>Some review.</review>
+				</book>
+				<secondcategory>
+					<paper id="bk102">
+						<test2>Ralls, Kim</test2>
+						<description>A former architect battles corporate zombies, 
+						an evil sorceress, and her own childhood to become queen 
+						of the world.</description>
+					</paper>
+				</secondcategory>
+				</catalog>`,
+		}}
+
+		for i := 0; i < b.N; i++ {
+			_, err = p.Run(evt)
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+}
+
+func TestXMLToDocumentID(t *testing.T) {
+	log := logp.NewLogger("decode_xml")
+
+	input := common.MapStr{
+		"message": `<catalog>
+						<book seq="10">
+							<author>William H. Gaddis</author>
+							<title>The Recognitions</title>
+							<review>One of the great seminal American novels of the 20th century.</review>
+						</book>
+					</catalog>`,
+	}
+
+	config := common.MustNewConfigFrom(map[string]interface{}{
+		"fields":      []string{"message"},
+		"document_id": "catalog.book.seq",
+		"target":      "message",
+	})
+
+	p, err := New(config)
+	if err != nil {
+		log.Error("Error initializing decode_xml")
```

Comment:
Avoid relying on `logp` in tests. If you need to log something from a tests use `t.Log`. But in this case you already have `t.Fatal` so just remove it.